### ISSUE TITLE
Update of location mechanism

### ISSUE THREAD

**Issue by svasylenko** [LSCOIT/LegalNavigator#855]
New location search mechanism to be created for Topics and any other resources. Currently search by topics is done only on state level.

New approach: to implement bottom-up search, i.e. firstly we look at ZIP (if known for current page visitor), if no results (or ZIP is unknown), look with City - if no -> up to County - if no -> State.

Note: since tables are partitioned by State - to consider doing pre-search by state and do main search after inside result.

**Comment by KellyeDC-zz** [comment]
Testing On Hold, as we continue to flush out requirements for geocoding.

**Comment by grawdon** [comment]
@KellyeDC How do I test this? I don't see an option to enter my zip code. It only lets me update by state. Even if I could enter a zip code, I won't know which topic areas have content geocoded down to the zip code.

**Comment by KellyeDC** [comment]
@grawdon We've been able to test this quite a bit.  It works fine now for zip code, but we have a few issues with county.  Do you want to keep this open or create a specific defect for county and linked it with this and the county requirement?

**Comment by grawdon** [comment]
@KellyeDC I would create a specific defect for county but remind them to test to be zip code is still working after county is fixed. As I think about it, I wonder if we should eliminate county all together for now. As Alex said, we want to encourage them to use zip code. We should see if they have any zip codes that overlap counties. If we could eliminate it, at least for now, it would simplify things.

**Comment by KellyeDC** [comment]
Makes sense @grawdon .  I will prioritize these location tasks for them next week.

**Comment by KellyeDC** [comment]
@svasylenko Have you had a chance to review this with issue #971 and #970 to make certain location mechanism is fixed?

**Comment by yevhenobertas** [comment]
@KellyeDC  both tickets above were verified by Glenn.
